src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.cs

```diff
@@ -25,37 +25,44 @@ public HostAvailabilityCheckMiddleware(RequestDelegate next, ILoggerFactory logg
             _scriptHostManager = scriptHostManager;
         }
 
-        public async Task Invoke(HttpContext httpContext)
+        public Task Invoke(HttpContext httpContext)
         {
-            if (_scriptHostManager.State != ScriptHostState.Offline)
+            // Slow path with async/await for spin-up only
+            static async Task InvokeAwaited(HttpContext context, RequestDelegate next, ILogger<HostAvailabilityCheckMiddleware> logger, IScriptHostManager scriptHostManager)
```

Comment:
I'd personally like to see this as a separate method... it's typically what we do throughout the codebase although there's no real concrete rules we follow.